components/button/include/iot_button.h

```diff
@@ -87,12 +87,14 @@ esp_err_t iot_button_delete(button_handle_t btn_handle);
  * @param btn_handle A button handle to register
```

Comment:
Re the commit log, I think you are trying to allow passing user data through to callbacks.